### DISCUSSION TITLE
Would this pool support older Bitcoin/Litecoin forks that don't have ZMQ integrated?

### DISCUSSION THREAD

**Discussion by BenzMo** [dreams-money/merged-mining-pool#10] [General]
Would this pool support older Bitcoin/Litecoin forks that don't have ZMQ integrated?

**Comment by dreams-money** [comment]
No.  You absolutely need ZMQ.

**Reply by BenzMo** [reply]
I managed to get a Python script that polls the old rpc method at regular intervals and converts it into ZMQ that the pool can understand. Several more issues resolved later I managed to successfully find and broadcast a block! 

But I have stumbled at what looks like the last hurdle.. the RewardTo did not submit/generate properly, it shows as 'Unknown' on the block explorer. Maybe something to do with the parameters/coinbase generation or older RPC method/structure for submitblock? Anyway this part seems too technical for me so will probably have to admit defeat and ask for a devs help.

**Reply by dreams-money** [reply]
That's understandable.   I'd recommend that your fork merge in some of the latest Litecoin updates.

**Reply by BenzMo** [reply]
I figured it out, finally! I think the pool looks to get `scriptPubKey` by calling the `validateaddress` rpc but for the older fork I'm using this method/command doesn't contain that variable. I decided to generate it manually and hardcode it. Successfully found the first primary coin block with correct reward address! I haven't tested on the testnet to see if Doge will work, fingers crossed it should.

**Reply by dreams-money** [reply]
Interesting.   So you're creating the scriptPubKey w/o the RPC?

**Reply by BenzMo** [reply]
Still using the RPC (I think, technically, via command line/terminal) but manually and with different functions/commands, using `createrawtransaction` then taking the resulting tx hex and using `decoderawtransaction` which produces the scriptPubKey variable.
